Replace tmp file cert/key upload with memory buffer upload.

jlongstaf opened the pull request:
@<reviewer_id>
#### What issues does this address?
Fixes #127 

#### What's this change do?

#### Where should the reviewer start?
ssl_profile.py

#### Any background context?
New method to pass in bytes instead of file name was added to f5-sdk. Replaced
creating tmp file with byte method to keep it in-memory.

Issues:
Fixes #127

Problem: Replace creating a tmp file for cert/key upload with
string buffer.

Analysis: Used new interface in f5-sdk to keep cert/key data
in memory.

Tests: ssl_profile test.

richbrowne commented:
See if there is a way to clear out the cert and keyfile memory buffers.  Not really sure how this might be done with python object.

swormke commented on f5_openstack_agent/lbaasv2/drivers/bigip/ssl_profile.py:
```diff
@@ -39,42 +38,33 @@ def create_client_ssl_profile(bigip, name, cert, key, sni_default=False):
+            param_set = {}
+            param_set['name'] = certfilename
+            param_set['from-local-file'] = \
+                '/var/config/rest/downloads/' + certfilename
```
probably want to use the os.path.join for this type of thing.

https://docs.python.org/2/library/os.path.html

jlongstaf commented:
@richbrowne: After investigating erasing memory it seems that there is not much we can do. The bottom line is that strings in Python are immutable and short of custom C code, the best we can do is to be aggressive in ensuring that the cert/key variables are deleted. See http://stackoverflow.com/questions/728164/securely-erasing-password-in-memory-python for a good summary of the problem.

I added  try block and del of cert and key to ensure that the memory is marked for garbage collection. Also, I added @swormke suggestion to used join() for the path.
